bfsml.sml

```diff
@@ -7,25 +7,31 @@ struct
 
 fun interpret (instr) =
     let
-	val datarr = Array.array(65536,chr 0)
+	val datarr = Array.array(65536, 0w0 : Word8.word)
 	val dtp = ref 0
+	fun load () = Array.sub(datarr, !dtp)
+	fun store v = Array.update(datarr, !dtp, v)
+	fun read () = case TextIO.input1(TextIO.stdIn) of
+			  NONE => 0wxff (* to be compatible with some interpreters *)
```

Comment:
What is the purpose of this case clause? If we read a None, doesn't that mean we reach the end of the file and we should exit the program?

Reply:
This does not mean the end of the program but the end of the input.  The program may be stuff at the end of the input.
Actually there is no convention on what value should be used on EOF.  I choose 0xff only because C's `getchar` returns -1 on EOF.  Some implementations use 0, or leave the cell unchanged.